lib/stdlib/src/binary.erl

```diff
@@ -363,6 +362,85 @@ get_opts_replace([{insert_replaced,N} | T],{Part,Global,_Insert}) ->
 get_opts_replace(_,_) ->
     throw(badopt).
 
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+%% Hex encoding functions
+%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
+-spec encode_hex(Bin) -> Bin2 when
+      Bin :: binary(),
+      Bin2 :: binary().
+encode_hex(Bin) when is_binary(Bin) ->
+    encode_hex(Bin, [lowercase]);
+encode_hex(Bin) ->
+    badarg_with_info([Bin]).
+
+-spec encode_hex(Bin, Options) -> Bin2 when
+      Bin :: binary(),
+      Options :: [Option],
+      Option :: lowercase | uppercase,
+      Bin2 :: binary().
+encode_hex(Bin, Options) when is_binary(Bin), is_list(Options) ->
+    case proplists:get_bool(uppercase, Options) of
+        false ->
+            encode_hex(Bin, fun enc_hex_digit_lower/1, <<>>);
+        true ->
+            encode_hex(Bin, fun enc_hex_digit_upper/1, <<>>)
+    end;
+encode_hex(Bin, Options) ->
+    badarg_with_info([Bin, Options]).
+
+-spec encode_hex(Bin, Enc16, Acc) -> Bin2 when
+      Bin :: binary(),
+      Enc16 :: fun((0..15) -> byte()),
+      Acc :: binary(),
+      Bin2 :: Acc.
+encode_hex(<<>>, _Enc16, Acc) ->
+    Acc;
+encode_hex(<<A0:4, B0:4, Rest/binary>>, Enc16, Acc) ->
+    A = Enc16(A0),
+    B = Enc16(B0),
+    encode_hex(Rest, Enc16, <<Acc/binary, A, B>>).
+
+-spec enc_hex_digit_upper(0..15) -> byte().
+enc_hex_digit_upper(Char) when Char =< 9 ->
+    Char + $0;
+enc_hex_digit_upper(Char) when Char =< 15 ->
```

Comment:
The `when Char =< 15` guard is redundant and should IMO be dropped for performance reasons.

Reply:
After benchmark, it appears that deleting the clause does not significantly improve performance. Without real performance improvement, I prefer to keep code readable. Don't hesitate if you have benchmarks to support your opinion.